### PR TITLE
[docs][1.5] update RPC docs to reflect correct use of dist_autograd backwards and dist_optim step()

### PR DESCRIPTION
**Pull request by rohan-varma**
- Clarify that `torch.distributed.autograd.backwards()` does not use the current thread local autograd context, instead it looks it up based on the context_id passed in
- Clarify the same for `torch.distributeed.optimizer.optim.step()`

### CODE REVIEW

File: torch/distributed/optim/optimizer.py [hunk 1]
```diff
@@ -128,8 +128,9 @@ def step(self, context_id):
 
         This will call :meth:`torch.optim.Optimizer.step` on each worker
         containing parameters to be optimized, and will block until all workers
-        return. The current distributed autograd
-        :class:`~torch.distributed.autograd.context` will be used globally.
+        return. The provided ``context_id`` will be used to retrieve the
```
Comment (osalpekar) [inline torch/distributed/optim/optimizer.py]: nit: Maybe slightly change the wording on this sentence? For example `The provided ``context_id`` will be used to retrieve the corresponding :class:~torch.distributed.autograd.context that contains the gradients to be applied on the parameters.`(or something similar?)

**Review by osalpekar — APPROVED** [review]
LGTM!
